spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java

```diff
@@ -107,6 +116,19 @@ public void init() {
 		}
 	}
 
+	@After
+	public void cleanup() {
+		try {
+			Seq<String> topics = ZkUtils.getAllTopics(kafkaTestSupport.getZkClient());
+			for (String topic: scala.collection.JavaConversions.seqAsJavaList(topics)) {
+				AdminUtils.deleteTopic(kafkaTestSupport.getZkClient(), topic);
+			}
+		}
+		catch (Exception e) {
+			log.error(e.getMessage());
```

Comment:
No cleanup is not needed if we run things embedded (that being said, we shouldn't run embedded by default by hardcoding the environment - we can just change the default in the rule). We can use the rule for checking whether we're running embedded or external. I guess for deciding what topics have been created we can do the same thing that Rabbit does, and record them in the test binder.